internal/core/software_renderer/fixed.rs

```diff
@@ -109,6 +109,17 @@ impl<T: core::ops::Mul<Output = T>, const SHIFT: usize> core::ops::Mul<T> for Fi
     }
 }
 
+impl<T: core::ops::Mul<Output = T>, const SHIFT: usize> core::ops::Mul<Fixed<T, SHIFT>>
+    for Fixed<T, SHIFT>
+where
+    T: core::ops::Shr<usize, Output = T>,
+{
+    type Output = Self;
+    fn mul(self, rhs: Fixed<T, SHIFT>) -> Self::Output {
+        Self(self.0.mul(rhs.0) >> SHIFT)
```

Comment:
I wonder if there's a way to get from `T` to a `U` that has double the number of bits, so that we avoid overflows. So from u16 to u32.

Reply:
maybe this:

```suggestion
        Self(self.0.mul(rhs.0 >> SHIFT) + self.0.mul(rhs.0 & ((1 << SHIFT)-1)) >> SHIFT)
```

Reply:
Clever :). Fine with me, please commit. The other idea I just had was to say: "screw it" and just cast to u64, multiply there, and then go back to T. Anyway, if you're ok with your suggestion as well, please press "commit"  :)

Reply:
I pressed commit but i realized it won't compile because it does `&` and `-` which is not in the trait.

Reply:
No problem, I'll fix it up - thanks :)